pkg/exporter/exporter.go

```diff
@@ -43,58 +46,66 @@ func New(clusterClient clientset.Interface, apiClient *client.APIClient, recorde
 }
 
 func (e *Exporter) Run(ctx context.Context, configMapName, region string, patrolInterval time.Duration, stopChan <-chan struct{}) {
+	//delete old configmap if any
```

Comment:
Why do you delete it first? You need to read the existing old data in case you will not be able to get the new data in the freshdata call.